### PR TITLE
VW button spam: fix safety and add tests

### PR DESCRIPTION
**Pull request by rbiasini**

### CODE REVIEW

**Comment by rbiasini** [comment]
@jyoung8607 , how does it look?

**Comment by jyoung8607** [comment]
> @jyoung8607 , how does it look?

It looks good at first glance. I'll take it for a drive tomorrow morning to verify.

**Comment by rbiasini** [comment]
@jyoung8607 I think this should be good to go: 
- [*] added gas press safety check and 
- [*] added button spam safety check
- [*] made nomenclature consistent

File: board/safety/safety_volkswagen.h [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-const int VW_MAX_RATE_UP = 4;               // 5.0 nm/s available rate of change from the steering rack
-const int VW_MAX_RATE_DOWN = 10;            // 5.0 nm/s available rate of change from the steering rack
-const int VW_DRIVER_TORQUE_ALLOWANCE = 80;
-const int VW_DRIVER_TORQUE_FACTOR = 3;
-
-struct sample_t vw_torque_driver;           // last few driver torques measured
-int vw_rt_torque_last = 0;
-int vw_desired_torque_last = 0;
-uint32_t vw_ts_last = 0;
+const int VOLKSWAGEN_MAX_STEER = 250;               // 2.5 nm
```
Comment (jyoung8607) [inline board/safety/safety_volkswagen.h]: Proposed comment touchup: // 2.5 nm (EPS side max of 3.0nm with fault if violated)

File: board/safety/safety_volkswagen.h [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+const int VOLKSWAGEN_MAX_RT_DELTA = 75;             // 4 max rate up * 50Hz send rate * 250000 RT interval / 1000000 = 125 ; 125 * 1.5 for safety pad = 75
```
Comment (jyoung8607) [inline board/safety/safety_volkswagen.h]: Proposed comment touchup: // 4 max rate up * 50Hz send rate * 250000 RT interval / 1000000 = 50 ; * 1.5 for safety pad = 75
(fixes an outdated figure in the calculation)

File: board/safety/safety_volkswagen.h [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
+const uint32_t VOLKSWAGEN_RT_INTERVAL = 250000;     // 250ms between real time checks
+const int VOLKSWAGEN_MAX_RATE_UP = 4;               // 5.0 nm/s available rate of change from the steering rack
```
Comment (jyoung8607) [inline board/safety/safety_volkswagen.h]: Proposed comment touchup: // 2.0 nm/s rate of change (EPS side delta-limit of 5.0nm/s)
Fixes a factually incorrect/outdated figure

File: board/safety/safety_volkswagen.h [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+const int VOLKSWAGEN_MAX_RATE_DOWN = 10;            // 5.0 nm/s available rate of change from the steering rack
```
Comment (jyoung8607) [inline board/safety/safety_volkswagen.h]: // 5.0 nm/s rate of change (EPS side delta-limit of 5.0nm/s)

File: board/safety/safety_volkswagen.h [hunk 5]
```diff
@@ -93,16 +105,25 @@ static int volkswagen_tx_hook(CAN_FIFOMailBox_TypeDef *to_send) {
...
     }
 
     if (violation) {
       tx = 0;
     }
   }
 
+  // FORCE CANCEL: ensuring that only the cancel button press is sent when controls are off.
+  // This avoids unintended engagements while still allowing resume spam
+  if ((addr == MSG_GRA_ACC_01) && !controls_allowed && (bus == 2)) {
```
Comment (jyoung8607) [inline board/safety/safety_volkswagen.h]: Proposed for clarity:
```  if ((bus == 2) && (addr == MSG_GRA_ACC_01) && !controls_allowed) {```
And similar for bus 0/HCA_01 earlier in the function.

File: board/safety/safety_volkswagen.h (hunk 1 above)
Comment (rbiasini) [reply board/safety/safety_volkswagen.h]: ok

File: board/safety/safety_volkswagen.h (hunk 2 above)
Comment (rbiasini) [reply board/safety/safety_volkswagen.h]: ok

File: board/safety/safety_volkswagen.h (hunk 4 above)
Comment (rbiasini) [reply board/safety/safety_volkswagen.h]: yeah, it felt like there was a built-in limit when testing

File: board/safety/safety_volkswagen.h (hunk 5 above)
Comment (rbiasini) [reply board/safety/safety_volkswagen.h]: ok

Comment (rbiasini) [reply board/safety/safety_volkswagen.h]: I'll keep the bus check on HCA_01 since it's not necessary, as we discussed

**Pull request merged**